libnmstate/nm/route.py

```diff
@@ -210,3 +217,94 @@ def get_static_gateway_iface(family, iface_routes):
             if route[Route.DESTINATION] == destination:
                 return iface_name
     return None
+
+
+def get_routing_rule_config(acs_and_ip_profiles):
+    rules = []
+    for (_, ip_profile) in acs_and_ip_profiles:
+        for i in range(ip_profile.get_num_routing_rules()):
+            nm_rule = ip_profile.get_routing_rule(i)
+            rules.append(_nm_rule_to_info(nm_rule))
+
+    return rules
+
+
+def _nm_rule_to_info(nm_rule):
+    info = {
+        RouteRule.IP_FROM: _nm_rule_get_from(nm_rule),
+        RouteRule.IP_TO: _nm_rule_get_to(nm_rule),
+        RouteRule.PRIORITY: nm_rule.get_priority(),
+        RouteRule.ROUTE_TABLE: nm_rule.get_table(),
+    }
+    cleanup_keys = []
+    for key, val in info.items():
+        if not val:
+            cleanup_keys.append(key)
+    for key in cleanup_keys:
+        del info[key]
+
+    return info
+
+
+def _nm_rule_get_from(nm_rule):
+    if nm_rule.get_from():
+        return _tuple_to_ip_addr_full(
+            nm_rule.get_from(), nm_rule.get_from_len()
+        )
+    return None
+
+
+def _nm_rule_get_to(nm_rule):
+    if nm_rule.get_to():
+        return _tuple_to_ip_addr_full(nm_rule.get_to(), nm_rule.get_to_len())
+    return None
+
+
+def _tuple_to_ip_addr_full(ip_addr, ip_addr_len):
+    return '{}/{}'.format(ip_addr, ip_addr_len)
+
+
+def _ip_addr_full_to_tuple(ip_addr_full, family):
```

Comment:
Can ipaddress stdlib help you here?

Or maybe it can be generalized somehow and place in iplib module? It shouts "unit test me".